### DISCUSSION TITLE
How to pass a parameter that's not in the key

### DISCUSSION THREAD

**Discussion by BlackMarq20** [TanStack/query#691] [General] [answered]
I have a token that I need to pass to each query function, however I don't want that token to be a part of the key. I don't see in the current documentation how this is achieved without providing it in the key, however I do see older discussions around it however they no longer seem to work. Is this possible?

**Comment by tannerlinsley** [comment]
Just in-line in the query function after the key.

**Reply by BlackMarq20** [reply]
Are you saying to create an anonymous function?

`useQuery(‘key’, (a) => queryFn(a))`

**Comment by tannerlinsley** [comment, marked as answer]
Indeed! Pass the keys through that you need, then tack on any extra arguments your function needs.

**Reply by bdrazen** [reply]
Am I crazy or did the api used to support an optional second array param for this use case?
